DirectX-refactoring

MMTsallagova opened the pull request:
DirectX Initialization refactoring:
- Create new Graphics class and add new function for DirectX Initialization

MMTsallagova committed:
DirectX Initialization refactoring

DirectX Initialization refactoring:
- Create new Graphics class and add new function for DirectX Initialization

MMTsallagova committed:
DirectX Initialization refactoring

DirectX Initialization refactoring:
- transferring variables for rendering from GameClass to GraphicsClass
- edit links

nikchernyakov commented on Engine/Game.cpp:
```diff
@@ -3,12 +3,16 @@
 
 Game::Game(HINSTANCE hInstance, WNDCLASSEX wc)
 {   
+	graphics = new Graphics(this);
```
Почему создание объекта и его инициализация в разных местах кода?
Сделай, чтобы они вместе были, а от остального кода пустой строкой отдели

nikchernyakov commented on Engine/Game.h:
```diff
@@ -25,24 +26,19 @@ class Game
-	ID3D11DepthStencilView* depthStencilView = NULL;          // ������ ����, ����� ������
-
 	InputDevice* inputDevice;
+	ID3DUserDefinedAnnotation* annotation;
```
А с этим у тебя в итоге не получилось перенести?

nikchernyakov commented on Engine/Game.h:
```diff
@@ -25,24 +26,19 @@ class Game
 	Camera* camera;
 	MeshRenderer* meshRenderer;
-	
+	Graphics* graphics;
```
Верни это в protected блок, а для доступа сделай метод const GetGraphics()

nikchernyakov commented on Engine/Graphics.cpp:
```diff
@@ -0,0 +1,96 @@
+
+	game->graphics->context->RSSetState(rastState);
+
+	D3D11_TEXTURE2D_DESC descDepth;           // ��������� � �����������
```
Переведи плиз комменты на английский, а то они не отображаются на гите, да и по код стайлу неправильно

nikchernyakov commented on Engine/Graphics.h:
```diff
@@ -0,0 +1,28 @@
+public:
+    Graphics(Game* currentGame);
+	
+	ID3D11Device* device;
```
У тебя тут отступ какой-то лишний по сравнению с конструктором, где неправильно?

nikchernyakov commented on Engine/Graphics.h:
```diff
@@ -0,0 +1,28 @@
+	IDXGISwapChain* swapChain;
+	ID3D11RenderTargetView* rtv;
+
+	ID3D11Texture2D* depthStencil = NULL;             // �������� ������ ������
```
Переведи плиз комменты на английский, а то они не отображаются на гите, да и по код стайлу неправильно

nikchernyakov commented on Engine/MeshObject.cpp:
```diff
@@ -25,7 +25,7 @@ MeshObject::MeshObject(Game* game, Transform* transform, MeshData* meshData, Sha
 	sd.pSysMem = meshData->GetVertices().data();
 
-	hr = game->device->CreateBuffer(
+	hr = game->graphics->device->CreateBuffer(
```
Чтобы не громоздить эту кучу соплей из обращений, можно в начале в конструкторе сохранить в переменную graphics. а потом к ней внутри обращаться

nikchernyakov commented on Engine/Texture.cpp:
```diff
@@ -5,7 +5,7 @@
 Texture::Texture(Game* game, const wchar_t* texturePath) : m_game(game)
 {
 	// �������� �������� �� �����
```
Переведи плиз комменты на английский, а то они не отображаются на гите, да и по код стайлу неправильно

MMTsallagova committed:
DirectX Initialization refactoring

- Translation
- Add  graphics object in protected
- Edit links
- Add GetGraphics();

MMTsallagova committed:
DirectX Initialization refactoring

- Fix annotation
- Add Getters

nikchernyakov commented on Engine/Graphics.cpp:
```diff
@@ -0,0 +1,132 @@
+	context->OMSetRenderTargets(1, &renderTargetView, depthStencilView);
+
+
+    return false;
```
Только сейчас заметил, а зачем здесь false возвращается?
Давай тогда уже true возвращать хотя бы, чтобы потом false возвращать в случае ошибки

nikchernyakov commented on Engine/Graphics.cpp:
```diff
@@ -0,0 +1,132 @@
+	/*Using the filled-in description structure and texture, create depth buffer object*/
+	device->CreateDepthStencilView(depthStencil, &descDSV, &depthStencilView);
+
+
```
Раз всё равно менять этот файл, почини количество пустых строк
Чтобы не было больше одной пустой строки разрыва)

nikchernyakov commented on Engine/Graphics.cpp:
```diff
@@ -0,0 +1,132 @@
+	descDSV.ViewDimension = D3D11_DSV_DIMENSION_TEXTURE2D;
+	descDSV.Texture2D.MipSlice = 0;
+
+	/*Using the filled-in description structure and texture, create depth buffer object*/
```
Для комментариев в одну строку лучше использовать // и делай пробел между этими символами и текстом

nikchernyakov commented on Engine/MeshObject.cpp:
```diff
@@ -76,20 +77,20 @@ MeshObject::MeshObject(Game* game, Transform* transform, MeshData* meshData, Sha
 		);
-	game->context->IASetIndexBuffer(pIndexBuffer.Get(), DXGI_FORMAT_R32_UINT, 0u);
-	game->context->IASetPrimitiveTopology(D3D10_PRIMITIVE_TOPOLOGY_TRIANGLELIST);
+    graphics->GetContext()->IASetIndexBuffer(pIndexBuffer.Get(), DXGI_FORMAT_R32_UINT, 0u);
```
Тут отступ сломался

nikchernyakov commented on Engine/Texture.cpp:
```diff
@@ -4,26 +4,28 @@
-	auto hr = DirectX::CreateWICTextureFromFile(m_game->device, texturePath, nullptr, &g_pTextureRV, 0);
+	graphics = game->GetGraphics();
+
+	/*Loading a texture from file*/
```
Тут тоже замечание по поводу комментария

nikchernyakov reviewed (changes requested):
Сорян))

MMTsallagova committed:
Add code style

- Correct code style
